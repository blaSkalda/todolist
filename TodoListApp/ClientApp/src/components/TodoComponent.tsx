import * as React from 'react';
import { connect } from 'react-redux';
import { ApplicationState } from '../store';
import * as TodoTasksService from '../store/TodoTasksService';
import Error from './utilities/ErrorComponent';
import TodoCard from './utilities/TodoCardComponent';
import TodoForm from './utilities/TodoFormComponent';

// At runtime, Redux will merge together...
type TodoTasksProps =
    TodoTasksService.TodoTasksState // ... state we've requested from the Redux store
    & typeof TodoTasksService.actionCreators; // ... plus action creators we've requested

class Todo extends React.PureComponent<TodoTasksProps> {

    // This method is called when the component is first added to the document
    public componentDidMount() {
        this.ensureDataFetched();
    }

    public componentDidUpdate() {

    }

    public render() {
        const todoTask: TodoTasksService.TodoTask = {
            name: '',
            priority: 0,
            status: 0
        };

        return (
            <React.Fragment>
                {this.props.error && <Error message={this.props.error} />}
                {this.renderTable()}
                <TodoForm todoTask={todoTask} {...this.props} />
            </React.Fragment>
        );
    }

    private ensureDataFetched() {
        this.props.requestTodoTasks();
    }

    private renderTable() {
        return (
            <div className="card-columns">
                {this.props.todoTasks.map((todoTask: TodoTasksService.TodoTask) =>
                    <TodoCard key={todoTask.name} todoTask={todoTask} {...this.props} />
                )}
            </div>
        );
    }
}

export default connect(
    (state: ApplicationState) => state.todoTasks, // Selects which state properties are merged into the component's props
    TodoTasksService.actionCreators // Selects which action creators are merged into the component's props
)(Todo as any);
