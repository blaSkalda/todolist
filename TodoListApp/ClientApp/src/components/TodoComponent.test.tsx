import * as React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme'

import TodoComponent from './TodoComponent';

it('renders TodoComponent', () => {
    const storeFake = (state: any) => ({
        default: () => { },
        subscribe: () => { },
        dispatch: () => { },
        getState: () => ({ ...state })
    });
    const store = storeFake({
        todoTasks: {
            todoTasks: []
        }
    }) as any;

    const wrapper = mount(<Provider store={store}><TodoComponent /></Provider>);
    expect(wrapper).toMatchSnapshot();
});
