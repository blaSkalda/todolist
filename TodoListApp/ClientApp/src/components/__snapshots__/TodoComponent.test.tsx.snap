// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TodoComponent 1`] = `
<Provider
  store={
    Object {
      "default": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Todo)>
    <Todo
      cancelTodoTask={[Function]}
      createTodoTask={[Function]}
      deleteTodoTask={[Function]}
      editTodoTask={[Function]}
      requestTodoTasks={[Function]}
      todoTasks={Array []}
      updateTodoTask={[Function]}
      updateTodoTaskStatus={[Function]}
    >
      <div
        className="card-columns"
      />
      <TodoForm
        cancelTodoTask={[Function]}
        createTodoTask={[Function]}
        deleteTodoTask={[Function]}
        editTodoTask={[Function]}
        requestTodoTasks={[Function]}
        todoTask={
          Object {
            "name": "",
            "priority": 0,
            "status": 0,
          }
        }
        todoTasks={Array []}
        updateTodoTask={[Function]}
        updateTodoTaskStatus={[Function]}
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          Create
        </button>
      </TodoForm>
    </Todo>
  </Connect(Todo)>
</Provider>
`;
